### PR TITLE
[CTCTRADERS-1366] Loop through adding guarantees flow

### PR DESCRIPTION
**Pull request by ericohwotu**
Updated the flow to loop through for adding guarantees.  In this PR only the flow and content has been updated there are other tickets to update the domain model and the xml model.

### CODE REVIEW

**Comment by ericohwotu** [comment]
test this please

**Review by ashwinirudrappa — APPROVED** [review]

File: app/controllers/guaranteeDetails/AddAnotherGuaranteeController.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    with I18nSupport
+    with NunjucksSupport {
+
+  def allowMoreGuarantees(ua: UserAnswers): Boolean =
+    ua.get(DeriveNumberOfGuarantees).getOrElse(0) < AddAnotherGuaranteePage.maxAllowedGuarantees
+
+  def onPageLoad(lrn: LocalReferenceNumber): Action[AnyContent] = (identify andThen getData(lrn) andThen requireData).async {
+    implicit request =>
+      val form = formProvider(allowMoreGuarantees(request.userAnswers))
+
```
[the remark this answers is not in the document]
Comment (ericohwotu) [reply app/controllers/guaranteeDetails/AddAnotherGuaranteeController.scala]: done

File: app/controllers/guaranteeDetails/ConfirmRemoveGuaranteeController.scala [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport {
+
+  private val form     = formProvider()
+  private val template = "guaranteeDetails/confirmRemoveGuarantee.njk"
+
+  def onPageLoad(lrn: LocalReferenceNumber, index: Index): Action[AnyContent] = (identify andThen getData(lrn) andThen requireData).async {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(ConfirmRemoveGuaranteePage) match {
```
[the remark this answers is not in the document]
Comment (ericohwotu) [reply app/controllers/guaranteeDetails/ConfirmRemoveGuaranteeController.scala]: fixed

File: app/forms/mappings/Mappings.scala [hunk 3]
```diff
@@ -20,13 +20,17 @@ import java.time.LocalDate
...
-import play.api.data.Forms.of
+import play.api.data.Forms.{ignored, of}
+import play.api.data.format.Formats.ignoredFormat
 
 trait Mappings extends Formatters with Constraints {
 
   protected def text(errorKey: String = "error.required", args: Seq[Any] = Seq.empty): FieldMapping[String] =
     of(stringFormatter(errorKey, args))
 
+  protected def mandatoryIfBoolean(condition: Boolean, requiredKey: String = "error.required"): FieldMapping[Boolean] =
```
Comment (ashwinirudrappa) [inline app/forms/mappings/Mappings.scala:31]: @ericohwotu Do we need this method, please check with the design what should be behavior when users reach the allowed limit?we used to show him the text that describes it not the radios.

Comment (ericohwotu) [reply app/forms/mappings/Mappings.scala:31]: The radios are not shown in this case but are still present on the form. So it only uses that answer if it allows more.

**Pull request merged**